src/ansys_sphinx_theme/__init__.py

```diff
@@ -110,9 +110,56 @@ def get_autoapi_templates_dir_relative_path(path: pathlib.Path) -> str:
         A string rerpesenting the relative path to the autoapi templates.
 
     """
-    return os.path.relpath(
-        str(AUTOAPI_TEMPLATES_PATH.absolute()), start=str(path.parent.absolute())
+    return os.path.relpath(str(AUTOAPI_TEMPLATES_PATH.absolute()), start=str(path.absolute()))
+
+
+def add_autoapi_theme_option(app: Sphinx) -> None:
+    """Add the autoapi template path to the theme options.
+
+    Parameters
+    ----------
+    app : ~sphinx.application.Sphinx
+        Application instance for rendering the documentation.
+    """
+    autoapi = app.config.html_theme_options.get("autoapi", {})
+    if not autoapi:
+        return
+    required_extensions = ["sphinx_jinja", "sphinx_design"]
+
+    for extension in required_extensions:
+        if extension not in app.config["extensions"]:
+            app.config["extensions"].append(extension)
+    AUTOAPI_OPTIONS = [
+        "members",
+        "undoc-members",
+        "show-inheritance",
+        "show-module-summary",
+        "special-members",
+    ]
+    app.add_css_file("https://www.nerdfonts.com/assets/css/webfont.css")
+    autoapi_template_dir = autoapi.get("templates", "")
+    autoapi_project_name = autoapi.get("project", "")
+
+    if not autoapi_template_dir:
+        autoapi_template_dir = get_autoapi_templates_dir_relative_path(app.confdir)
+
+    app.config["autoapi_template_dir"] = autoapi_template_dir
+
+    def prepare_jinja_env(jinja_env) -> None:
+        """Prepare the Jinja environment for the theme."""
+        jinja_env.globals["project_name"] = autoapi_project_name
+
+    # Set the autoapi options
+    app.config["autoapi_prepare_jinja_env"] = prepare_jinja_env
+    app.config["autoapi_type"] = autoapi.get("type", "python")
+    app.config["autoapi_root"] = autoapi.get("output", "api")
+    app.config["autoapi_own_page_level"] = autoapi.get("own_page_level", "module")
```

Comment:
This should be class by default:

```suggestion
    app.config["autoapi_own_page_level"] = autoapi.get("own_page_level", "class")
```